platform/openide.loaders/src/org/openide/loaders/TemplateWizard.java

```diff
@@ -244,7 +252,14 @@ public DataFolder getTemplatesFolder () {
     public DataFolder getTargetFolder () throws IOException {
         LOG.log(Level.FINE, "targetFolder={0} for {1}", new Object[] {targetDataFolder, this});
         if (targetDataFolder == null) {
-            throw new IOException(NbBundle.getMessage(TemplateWizard.class, "ERR_NoFilesystem"));
+            // Fix for [NETBEANS-3918] and [NETBEANS-3875]
+            File targetFolderFile = (File) getProperty(PROP_TARGET_FOLDER_HINT);
+            if(targetFolderFile != null) {
+                targetFolderFile.mkdirs();
+                setTargetFolder(DataFolder.findFolder(FileUtil.toFileObject(targetFolderFile)));
```

Comment:
Please note that `TemplateWizard` hasn't yet been using `java.io.File`. This would be the first time and I am not happy about it as it limits the usage of `TemplateWizard` to local disk. Could you consider alternative solution? Could you introduce
```java
public void setTargetFolder(Supplier<DataFolder> toBeCreatedFolder) {
}
```
e.g. give the code a chance to work with `DataFolder` and `FileObject` only, but delay creation of the actual folder as much as possible?